packages/core/src/sagas/stopSpeakActivityOnInputSaga.js

```diff
@@ -1,39 +1,39 @@
 import {
   put,
-  select,
-  take
+  takeEvery
 } from 'redux-saga/effects';
 
 import whileConnected from './effects/whileConnected';
 
-import markActivity from '../actions/markActivity';
-import stopSpeakingActivity from '../actions/stopSpeakingActivity';
 import { POST_ACTIVITY_PENDING } from '../actions/postActivity';
 import { SET_SEND_BOX } from '../actions/setSendBox';
 import { START_DICTATE } from '../actions/startDictate';
+import stopSpeakingActivity from '../actions/stopSpeakingActivity';
 
 export default function* () {
   yield whileConnected(function* () {
-    for (;;) {
-      yield take(
-        ({ payload, type }) => type === START_DICTATE
-        || (type === SET_SEND_BOX && payload.text && payload.via !== 'speech')
+    yield takeEvery(
+      ({ meta, payload, type }) => (
```

Comment:
inlining this takes away from the readability for me. can it be defined outside the function and given a good name?